mget.go

```diff
@@ -51,84 +46,24 @@ func (m *MGetRequest) SetIndex(index string) *MGetRequest {
 	return m
 }
 
-// MarshalJSON marshals the MGetRequest into the proper json expected by OpenSearch.
-func (m *MGetRequest) MarshalJSON() ([]byte, error) {
-	docs := make([]any, len(m.Docs))
-	for i, d := range m.Docs {
-		docReq := map[string]any{
-			"_id": d.ID(),
-		}
-
-		if d.Index() != "" {
-			docReq["_index"] = d.Index()
-		}
-
-		docs[i] = docReq
-	}
-
-	source := map[string]any{
-		"docs": docs,
-	}
-
-	return json.Marshal(source)
-}
-
-// Do executes the Multi-Get MGetRequest using the provided opensearch.Client.
-// If the request is executed successfully, then a MGetResponse with MGetResults will be returned.
-// An error can be returned if
-//
-//   - The request to OpenSearch fails
-//   - The results json cannot be unmarshalled
-func (m *MGetRequest) Do(ctx context.Context, client *opensearch.Client) (*MGetResponse, error) {
-	bodyBytes, jErr := json.Marshal(m)
-	if jErr != nil {
-		return nil, jErr
-	}
-
-	osResp, rErr := opensearchapi.MgetRequest{
-		Index: m.Index,
-		Body:  bytes.NewReader(bodyBytes),
-	}.Do(ctx, client)
-
-	if rErr != nil {
-		return nil, rErr
-	}
-
-	resp := &MGetResponse{
-		StatusCode: osResp.StatusCode,
-		Header:     osResp.Header,
-	}
-
-	var respBuf bytes.Buffer
-	if _, err := respBuf.ReadFrom(osResp.Body); err != nil {
-		return nil, err
-	}
-
-	if err := json.Unmarshal(respBuf.Bytes(), &resp); err != nil {
-		return nil, err
-	}
-
-	return resp, nil
-}
-
 // MGetResponse wraps the functionality of [opensearchapi.Response] by unmarshalling the response body into a
 // slice of MGetResults.
 type MGetResponse struct {
-	StatusCode int          `json:"-"`
-	Header     http.Header  `json:"-"`
-	Docs       []MGetResult `json:"docs,omitempty"`
+	Header     http.Header
```

Comment:
Pedantic nit: `http.Header` is a map (multiple values) so the field name should be `Headers` (plural)

Reply:
Totally agree that it should be plural but it shoulda been plural in the standard lib (https://pkg.go.dev/net/http#Header, https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/net/http/request.go;l=169) as well but since it's not, I'd suggest keeping it consistent, but I'm willing to be convinced otherwise.